### DISCUSSION TITLE
Sentry not showing source

### DISCUSSION THREAD

**Discussion by theahmadzai** [getsentry/sentry-react-native#3019] [Q&A]
Hi, I'm using sentry for react native I have integrated it with gradle it uploads the sourcemaps fine, but I'm getting this error on sentry UI.

![image](https://user-images.githubusercontent.com/9446563/235124934-d96d6c1b-6a10-4a52-8970-fbce9625fed1.png)

We have couple of white-label apps that have same source code but different version and id, I want to know what am I missing
I see that it uploads source maps correctly for each app and version to the senty I can see that 

![image](https://user-images.githubusercontent.com/9446563/235125272-279d5a21-4a88-4ccd-bf61-b7fe320b1b7f.png)

This is how clicking on uploaded source maps look 

![image](https://user-images.githubusercontent.com/9446563/235125546-f15ae711-345f-40f2-9aa1-d93c5db05ae8.png)

I also see some stuff uploaded to Debug Files section but ProGuard is empty

Thanks alot

**Comment by krystofwoldrich** [comment]
Hi,
I'm sorry for the late response,
At first glance, it looks like everything is uploaded correctly.

Does this warning happens with all the app or some of them are symbolicated correctly and only some of them are not?

Could you check specifically the mentioned release in the error, `com.tryperdiem.wh.breadsbakery@0.0.2+34`?
What does its release bundle look like?

**Reply by theahmadzai** [reply]
Thank you for your reply, It happens with all the apps not sure what you mean by looking into the release bundle

This is the complete screen shot of another example

![image](https://user-images.githubusercontent.com/9446563/235901196-931199c9-9a20-4e6a-9625-b1c297cf3278.png)

**Reply by krystofwoldrich** [reply]
I meant if you could check if this release, that the issue id from has uploaded artifacts. The easiest is to click on the release tag and then in the right column, there is a link to `X artifacts`.

**Reply by theahmadzai** [reply]
@krystofwoldrich 
For the same error above image, when I click on the release tag

![image](https://user-images.githubusercontent.com/9446563/236183056-b319f8ee-e156-4f60-8159-6635c6c5bdac.png)

then clicking on 2 artifacts take me here 

![image](https://user-images.githubusercontent.com/9446563/236183518-8b06f426-10da-4689-a3b6-ac8f99053cf0.png)

**Reply by theahmadzai** [reply]
The .bundle is an empty file btw but not the .map

**Reply by krystofwoldrich** [reply]
The empty bundle is expected for Hermes, as only the map is used for the symbolication.

How do you upload the source maps?

**Reply by theahmadzai** [reply]
Sentry plugin does it automatically I have followed the react native guide for sentry

**Reply by theahmadzai** [reply]
Could it be the issue that we have different apps under the same sentry project, but the codebase is the same for all apps apart from their version and identifier

**Reply by krystofwoldrich** [reply]
As long as the app can be identified by a `release name` and `dist` it's not a problem. Do you have a different dist for Andoird, iOS builds of the same release or do you have a release for Android and a release for iOS?

**Reply by theahmadzai** [reply]
The dist number should be the same for both Android and iOS but currently we only increment dist for Android also at the moment we are only uploading source maps when bundling Android.

Same source code example:-

Android
---
Package: com.tryperdiem.wh.perdiemcoffee
Version: 2.5.5
Dist: 237 (incremented each release)

iOS
---
Package: com.tryperdiem.wh.perdiemcoffee
Version: 2.5.5
Dist: 2 (same for all releases)

This is our sentry initialize
```javascript
Sentry.init({
  dsn: SENTRY_DSN,
  tracesSampleRate: 1.0,
  environment: ENV,
  attachScreenshot: true,
  attachStacktrace: true,
});
```

**Reply by krystofwoldrich** [reply]
Okay, thank you so the dist is not the problem.

And the stack trace is not symbolicated bot on Android and iOS for you?

**Reply by theahmadzai** [reply]
I don't know what does it mean but it looks like this 
![image](https://github.com/getsentry/sentry-react-native/assets/9446563/322c21c7-dbb6-402a-bad7-608acedef91d)

**Reply by krystofwoldrich** [reply]
Not symbolicated means you are not seeing the original unmodified functions and file names.
Are you still having this issue or did you manage to resolve this?

**Comment by Bobjwe2** [comment]
The issue likely stems from mismatched **release/version identifiers** or missing **ProGuard mapping files** in your Sentry setup.

### Checklist to Resolve:
1. **Ensure Release Matching:**  
   - Verify that the `release` and `dist` values in your React Native app and Sentry upload match exactly.  
     Example:  
     ```javascript
     Sentry.init({
       dsn: "YOUR_DSN",
       release: "com.example.app@1.0.0+1", // Ensure this matches Gradle configuration
       dist: "1",
     });
     ```

2. **Upload ProGuard Mapping Files:**  
   - If ProGuard is enabled, ensure mapping files are uploaded by adding this to your `app/build.gradle`:  
     ```groovy
     apply plugin: 'com.android.application'
     apply plugin: 'io.sentry.android.gradle'
     
     sentry {
         autoUploadProguardMapping = true
     }
     ```

3. **Check Debug Symbols:**  
   - Confirm uploaded source maps and debug symbols match the app version and are associated correctly in the Sentry UI.

4. **Unique Releases for White-Label Apps:**  
   - For each app, ensure `release` and `dist` values are unique, e.g.,  
     `com.appname1@1.0.0+1`, `com.appname2@1.0.0+1`.

5. **Verify Build Process:**  
   - Confirm your Gradle task (`app:assembleRelease` or equivalent) correctly uploads the source maps and ProGuard mappings.

Let me know if you need help troubleshooting any specific step!
